docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -12,261 +12,155 @@
-  Note that with the **integrated** approach we have deal with a mix of applications transactions EP and FPC-lifecycle EP.
-end note
-
-ref over User1, ERCC1
-  Create channel; each peer joins channel; each peer launches TLCC, which will  "imprint" itself with CH_id (content-addressable channel identifier, hash of gensis block)
+ref over ERCC1, ERCC2
+  It is assumed that: channel has already been created; ERCC has been installed.
+  Also: ERCC contains hard-coded the TLCC_MRenclave and the IAS Root Certificate.
 end ref
-|||
 
-ref over User1, ERCC1
-  Install and approve ERCC chaincode. ERCC defines TLCC_MR and IAS_PK in chaincode definition.
-  (alternatively, TLCC_MR and IAS_PK could be in channel configuration; crucial is that they are approved by humans (admins) on sufficiently many organizations)
-  ERCC also reads CH_id from ledger or, if that does not work, also embeds it into chaincode
-
-  see fpc-lifecycle-v2.puml
-end ref
-|||
-
-
-group enclave creation and attestation generation
-  
-  activate Admin
-
-  Admin -> Manager1 ++: create (CC_NAME, CC_VER, CH_name)
-  note right
-    CC_NAME: chaincode name
-    CC_VER: chaincode version
-    CH_NAME: channel name (needed by the ECC_Manager to cc2cc ECC
-  end note
-  note over Manager1 #FFaa00
+group enclave creation
+  note over Admin #FFaa00
     **MST**: we might want to replace (here and elsewhere) CC_VER with the sequence
     number from chain code definition as versions can be re-used in downgrades yet we
     might or might not want to re-use enclaves?
   end note
 
-  Manager1 -> Manager1 : get SPID and IAS credentials
-  note right: from local peer config
-
-  Manager1 -> ECC1 ++: create (SPID)
-  note left: this is a cc2cc call
-
-
-  ECC1 -> TLCC1 ++: getInfo
-  note over TLCC1
-    This information could also be retrieved elsewhere if easier. At this point in time
-    it does not have to be authenticated (and is anyway passed through untrusted ECC),
-    consistency will be verified later in registration (Remember that TLCC_MR is defined
-    and agreed by human consensus in the ERCC chaincode definition)
-  end note
-  return TLCC_MR, CH_id
-
-
-  |||
-
-  create participant "ECC_Enclave" as Enclave1 order 30 #99FF99
-  ECC1 -> Enclave1 ++: create (SPID, CC_NAME, CC_VER, Org_1, TLCC_MR, CH_id)
-  note left
-    this is a cgo call
-    Org_1 denotes the peer's MSP_ID
-  end note
+  Admin -> ECC1 ++: invoke create(SPID)
+
+  'ECC1 -> TLCC1 ++: getInfo
+  'note over TLCC1
+  '  This information could also be retrieved elsewhere if easier. At this point in time
+  '  it does not have to be authenticated (and is anyway passed through untrusted ECC),
+  '  consistency will be verified later in registration (Remember that TLCC_MR is defined
+  '  and agreed by human consensus in the ERCC chaincode definition)
+  'end note
+  'return TLCC_MR, CH_id
+
+  ECC1 -> Enclave1 ++: invoke create(SPID)
+
+  Enclave1 -> ECC1 ++: get_signed_proposal
```

Comment:
As already mentioned by me and marcus, it would be nicer to have explicit enclave lifecycle/mgmt functions in the edl rather than just splicing through the shim commands.  I.e., much closer to what we had in the old version than what you have now.  Certainly, to get the parameters for the calls we might use the shim functions you listed but the would be all inside the untrusted part of the ecc.  On one hand i think this makes understanding the protocol much easier and allow the reasoning of the security and also allows much more easily eventually tighter integration with dedicated fabric cli admin commands or alike.  


btw: given that you do not really need the signature and i think the shim can give you all values consumed, i would replace the get_signed_proposal call with individual getters for the required fields.  The signature here is really confusing when you look at it from a security protocol perspective as we don't really care about this signature at all ...

Reply:
> As already mentioned by me and marcus, it would be nicer to have explicit enclave lifecycle/mgmt functions in the edl rather than just splicing through the shim commands I.e., much closer to what we had in the old version than what you have now.

this was _necessary_ for the old version. In this version, it does not make much difference.

> Certainly, to get the parameters for the calls we might use the shim functions you listed but the would be all inside the untrusted part of the ecc. On one hand i think this makes understanding the protocol much easier and allow the reasoning of the security and also allows much more easily eventually tighter integration with dedicated fabric cli admin commands or alike. btw: given that you do not really need the signature and i think the shim can give you all values consumed, i would replace the get_signed_proposal call with individual getters for the required fields. The signature here is really confusing when you look at it from a security protocol perspective as we don't really care about this signature at all ...

This is more an implementation detail than design. Sure we can have a different call. The point is whether we should implement something additional when we already have something that works -- the `get_signed_proposal` is part of the fabric shim, which we are already supposed to expose, and which already returns the necessary information (though, as you say, more than that).

Reply:
your last point is exactly where i was heading it.  We need somewhere a protocol specification which leaves out irrelevant implementation details so one can easily see the overall flow and reason about the security.   The implementation details you added not only blur the bigger picture but imho also force certain implementation choices (e.g., they force a query-embedding of the management command and preclude a separate clean cli extensions).  Additionally, leaving low-level implementation details out of the UML also will increase the propability that the diagrams will also be uptodate over time

Reply:
I have tried to remove many details and to separate the management part.
Ok to remove the blur, and let's move on with the implementation -- the fastest one is clearly by embedding the management part in queries.

Ok to leave out implementation details, but there is nothing wrong (and it's actually highly desirable imho) if we update the diagrams over time as we learn!